pilot/pkg/model/context.go

```diff
@@ -441,6 +441,11 @@ func IsApplicationNodeType(nType NodeType) bool {
 	}
 }
 
+// IsEgressGateway checks that the proxy is an egress gateway.
+func IsEgressGateway(node *Proxy) bool {
+	return node.Type == Router && node.Metadata.Labels["istio"] == "egressgateway"
```

Comment:
@rshriram do we have a better way to detect egress? This doesn't seem like a solid solution

Reply:
Is there a concern besides detection code? I don't want to tie in some redesign of gateway labelling here.

Reply:
We donot enforce egress gateway to have this label, even ingress gateway can be used as a egress gateway.

Reply:
The concern here is that Egress gateway can have any label. istio=egressgateway is an  arbitrary one that happens to be the default

Reply:
I hear you. For better or worse, this is what we have, and we can make adjustments to egress policy & telemetry before deciding on what makes a gateway an egress gateway.